docker/Dockerfile.pdo-dev

```diff
@@ -119,57 +122,53 @@ RUN apt-get update \
- && sed -i '1,$s;${SGX_PACKAGES_PATH}/${PSW_PKG_NAME}/scripts/install.sh;# ${SGX_PACKAGES_PATH}/${PSW_PKG_NAME}/scripts/install.sh;' linux/installer/bin/install-sgx-psw.bin.tmpl \
- && make \
- && make sdk_install_pkg \
- && make psw_install_pkg \
- && cd .. \
- && echo "yes" | ./linux-sgx/linux/installer/bin/sgx_linux_x64_sdk_*.bin \
- && ./linux-sgx/linux/installer/bin/sgx_linux_x64_psw_*.bin \
- && rm -rf linux-sgx \
- && echo ". /opt/intel/sgxsdk/environment" >> /etc/profile.d/pdo.sh
-
-# ("Untrusted") OpenSSL
-WORKDIR /tmp
-RUN \
-    OPENSSL_MAJOR_VERSION=$(echo ${OPENSSL} | sed 's/\([^0-9.]\)*//g') \
-  && wget https://www.openssl.org/source/old/${OPENSSL_MAJOR_VERSION}/openssl-${OPENSSL}.tar.gz
-# && tar -zxvf openssl-${OPENSSL}.tar.gz \
-# && cd openssl-${OPENSSL}/ \
-# && ./config \
-# && THREADS=8 \
-# && make -j$THREADS \
-# && make test \
-# && make install -j$THREADS \
-# && ldconfig \
-# && ln -s /etc/ssl/certs/* /usr/local/ssl/certs/ \
-# && cd .. \
-# && rm -rf openssl-${OPENSSL}
-# Note: we do _not_ delete openssl-${OPENSSL}.tar.gz as we re-use it below ..
-
-
-# ("trusted") SGX OpenSSL
-# Note: This will compile in HW or SIM mode depending on the availability of
-# /dev/isgx and /var/run/aesmd/aesm.socket
-# Notes: there is no way to pass device /dev/isgx and socket (volume) /var/run/aesmd:/var/run/aesmd to docker
-# at build time, However, as we only build libraries here, the mode does not matter for the actual build
-# artifact and so we build with the safe SGX_MODE=SIM.  (It also means the build tests are run only in
-# simulator but we count on the sgxssl team hopefully releasing only versions which work on both cases :-)
-WORKDIR /tmp
-RUN git clone  --branch ${SGXSSL} https://github.com/intel/intel-sgx-ssl.git \
- && . /opt/intel/sgxsdk/environment \
- && (cd intel-sgx-ssl/openssl_source; mv /tmp/openssl-${OPENSSL}.tar.gz . ) \
- && (cd intel-sgx-ssl/Linux; make SGX_MODE=SIM DESTDIR=/opt/intel/sgxssl all test ) \
- && (cd intel-sgx-ssl/Linux; make install ) \
- && rm -rf /tmp/intel-sgx-ssl \
- && echo "export SGX_SSL=/opt/intel/sgxssl" >> /etc/profile.d/pdo.sh
+RUN wget https://download.01.org/intel-sgx/sgx-linux/${SGXSDK}/distro/ubuntu${UBUNTU_VERSION}/sgx_linux_x64_sdk_${SDKBIN}.bin
+RUN chmod +x sgx_linux_x64_sdk_${SDKBIN}.bin
+RUN echo -e "no\n/opt/intel" | ./sgx_linux_x64_sdk_${SDKBIN}.bin
+RUN echo ". /opt/intel/sgxsdk/environment" >> /etc/profile.d/pdo.sh
+
+# - LVI mitigations, needed to compile sgxssl requires a
+#   recent version of binutils (>= 2.32). Ubuntu 18.04 only
+#   has 2.30 but Intel ships binary distro for 2.32.51.20190719
+RUN wget "https://download.01.org/intel-sgx/sgx-linux/${SGXSDK}/as.ld.objdump.gold.r1.tar.gz"
+RUN mkdir sgxsdk.extras
+RUN cd sgxsdk.extras && \
+    tar -zxf ../as.ld.objdump.gold.r1.tar.gz
+RUN echo "export PATH=/opt/intel/sgxsdk.extras/external/toolset:${PATH}" >> /etc/profile.d/pdo.sh
+ENV PATH="/opt/intel/sgxsdk.extras/external/toolset:${PATH}"
```

Comment:
So, locally on my machine, I seem to have the right version of binutils. But I got a compiler error on Travis without these commands, because the lfence related gcc flags weren't recognized, so I added this back in.

Reply:
looking on my machines, even after `apt dist-upgrade`, i still have only `2.30-21ubuntu1~18.04.3`.  I suspect that you pulled it via some other repo (what does `apt-cache policy binutils` give you?).

If it is really not in 18.04 by default, then you will have to add the corresponding instructions from here also to the README!